lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -176,8 +188,130 @@ export default class BitcoinProcessor {
   }
 
   /**
-   * Trims the databases to the block prior to the block the given transaction is in.
-   * @returns The last fully processed block after trimming. `undefined` if all data are deleted after trimming.
+   * A faster version of process transactions that requires access to bitcoin data directory
+   * @param startingBlock the starting block to begin processing
+   */
+  private async fastProcessTransactions (startingBlock: IBlockInfo) {
+    const fileReader = new BitcoinFileReader(this.bitcoinDataDirectory!);
+    const blockFiles = fileReader.listBlockFiles().sort();
+    let fileIndex = blockFiles.length - 1;
+
+    const bestHeight = await this.bitcoinClient.getCurrentBlockHeight();
+    const bestHash = await this.bitcoinClient.getBlockHash(bestHeight);
+
+    let currentHash = bestHash;
+    let previousHash = undefined;
+    let numOfBlocksToProcess = bestHeight - startingBlock.height + 1;
+
+    const feeInfos = new Array(numOfBlocksToProcess).fill(undefined);
+
+    console.log(`Begin fast processing block ${startingBlock.height} to ${bestHeight}`);
+    // loop through files and process them until we process all blocks needed
+    while (numOfBlocksToProcess !== 0 && fileIndex >= 0) {
+      const blockFileName = blockFiles[fileIndex];
+      fileIndex--;
+      const blockFile = fileReader.readBlockFile(blockFileName);
+      const blockData = BitcoinRawDataParser.parseRawDataFile(blockFile);
+      // for each block within the current file, add fee info and write to db
+      for (let key in blockData) {
```

Comment:
You know what I'm going to say :) Can we break this down into smaller functions? I think it would more mantainable and testable if the while loop(s) can be 'read in english' (as much as possible off course). Kinda like:

````
...
while (numOfBlocksToProcess ...) {
  ...
  // my function names are only examples and are not accurate.
  const feeInfoFromFile = this.saveSidetreeTransactionsAndGetFeeInfo(fileName);
  this.removeInvalidFeeAndTransactionDbData(...); 
  ...
}
````